newIDE/app/src/UI/WikiSearchBar.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import * as React from 'react';
+import { I18n } from '@lingui/react';
+import { Trans, t } from '@lingui/macro';
+import MuiTextField from '@material-ui/core/TextField';
+import { indexName, searchClient } from '../Utils/AlgoliaSearch';
+import {
+  InstantSearch,
+  useInstantSearch,
+  useSearchBox,
+} from 'react-instantsearch-hooks';
+import SearchBarContainer from './SearchBarContainer';
+import Autocomplete from '@material-ui/lab/Autocomplete/Autocomplete';
+import { useDebounce } from '../Utils/UseDebounce';
+import { type GoToWikiCommand } from '../CommandPalette/CommandManager';
+import Window from '../Utils/Window';
+import { type AlgoliaSearchHit as AlgoliaSearchHitType } from '../Utils/AlgoliaSearch';
+import { AlgoliaSearchHit } from '../CommandPalette/CommandPalette/AutocompletePicker';
+
+type Props = {|
+  id?: string,
+|};
+
+const WikiSearchBar = ({ id }: Props) => {
+  const [autocompleteValue, setAutocompleteValue] = React.useState<string>('');
+  const [value, setValue] = React.useState<string>('');
+  const [isInputFocused, setIsInputFocused] = React.useState<boolean>(false);
+  const handleBlur = () => {
+    setIsInputFocused(false);
+    if (!value || value.trim() === '') {
+      setValue('');
+    }
+  };
+  const [
+    algoliaSearchStableStatus,
+    setAlgoliaSearchStableStatus,
+  ] = React.useState<'error' | 'ok'>('ok');
+
+  const { results, status } = useInstantSearch();
+  const { refine } = useSearchBox();
+
+  React.useEffect(
+    () => {
+      if (algoliaSearchStableStatus === 'ok' && status === 'error') {
+        setAlgoliaSearchStableStatus('error');
+      } else if (algoliaSearchStableStatus === 'error' && status === 'idle') {
+        setAlgoliaSearchStableStatus('ok');
+      }
+    },
+    [status, algoliaSearchStableStatus]
+  );
+
+  const launchSearch = useDebounce(() => {
```

Comment:
can we send an event when this is launched? to get some information about what is typed, and also how many times it is used